Directory.Build.targets

```diff
@@ -0,0 +1,41 @@
+ <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```

Comment:
@weshaggard @janvorli Could you take a look at this part of the change?  I'm not comfortable enough with the build system to say whether this is appropriate.  My concerns would be about this being Windows-specific and whether a new Directory.Build.targets files is too big a hammer to use for this case, but I don't know if those concerns are warranted.

Reply:
@weshaggard @janvorli Ping, have you had chance to review this PR

Reply:
I am sorry for not responding earlier, I was on vacation since Feb 2. Unfortunately, I have only a very light understanding of the msbuild projects as opposed to cmake project files and so I really cannot tell if this is the right way to do what we are trying to achieve or not.  
So I guess @weshaggard is the one who can decide here.

Reply:
In general I don't like having things in the common props/targets that are only intended to be used in one project (or solution in this case). I would suggest not special casing this and instead making it the default custom bin-placing logic for all build types.

Reply:
> I would suggest not special casing this and instead making it the default custom bin-placing logic for all build types.

It is my preference as well, however, I tried for this PR to have as limited impact on build system as possible.  I will convert it to default bin-placing logic for all build types.